src/modeling_library/distributions/beta.jl

```diff
@@ -12,7 +12,15 @@ const beta = Beta()
 
 function logpdf(::Beta, x::Real, alpha::Real, beta::Real)
     (x < 0 || x > 1 ? -Inf :
-    (alpha - 1) * log(x) + (beta - 1) * log1p(-x) - logbeta(alpha, beta) )
+        ((x == 0 && alpha <= 1) ?      #Special case to avoid infinite PDF at 0
+            (alpha - 1) * log(eps(typeof(x))) -
```

Comment:
If the user uses a constraint like `choicemap(:x => 0)`, there's a chance that `x` will be of type `Int`. It looks like this will cause `eps(typeof(x))` to error, but that's probably fine (the user can always use 0.0).